### PR TITLE
fix: record correct screenshots for popups

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
+ part of #253 
+ captures proper screenshots for new navigations done via popups and new tabs.
+ Listens for `page` events in each step and captures the screenshot of last navigation that happened in the step. 

### Tested using inline script for popups and navigation
```js
 step('Popup Landing Page', async () => {
  await page.goto(`http://localhost:8080/popup1.html`);
  await page.waitForSelector('text=Test popup page 1');
});
step('Open Popup', async () => {
  const [popup] = await Promise.all([
    page.waitForEvent('popup'),
    page.click('#linkWithPopup'),
  ]);
  // Test 2 - new tab (not new popup window)
  await Promise.all([
    context.waitForEvent('page'),
    page.click('#linkWithPopupNoNewWindow'),
  ]);
});

```

### Screenshots

Screen Shot 2021-07-21 at 17 00 12
Screen Shot 2021-07-21 at 17 00 02

### CODE REVIEW

File: src/cli.ts [hunk 1]
```diff
@@ -51,9 +51,16 @@ if (process.env.DEBUG === namespace || Boolean(options.debug)) {
...
+  const scriptFn = new Function(
+    'step',
+    'page',
+    'context',
+    'browser',
+    'params',
+    source
+  );
+  journey('inline', async ({ page, context, browser, params }) => {
+    scriptFn.apply(null, [step, page, context, browser, params]);
```
Comment (vigneshshanmugam) [inline src/cli.ts:63]: allows the use of `context` in inline journeys, was missing before.

File: src/core/runner.ts [hunk 2]
```diff
@@ -243,9 +244,19 @@ export default class Runner extends EventEmitter {
...
-      data.url ??= driver.page.url();
-      if (screenshots && screenshots !== 'off') {
-        await this.captureScreenshot(driver.page, step);
+      /**
+       * Capture screenshot for the newly created pages
+       * via popup or new windows/tabs
+       *
+       * Last open page will get us the correct screenshot
+       */
+      const pages = driver.context.pages();
```
Comment (vigneshshanmugam) [inline src/core/runner.ts:253]: Note: This handles if any popups/tabs are closed during the step. 

Example

```
// Test 1 - new window (popup)
const [popup] = await Promise.all([
  page.waitForEvent('popup'),
  page.click('#linkWithPopup'),
]);
await popup.close();
```

**Review by andrewvc — COMMENTED** [review]
This needs some tests. Also, after the PR is merged we should add this to the docs. It's not entirely obvious how this would work to a user. 

Also, if a user wants to switch the focus back to the original page how does that work? Looking at the code it looks like you'd need to close the popup. Is that right? If so, we should document that.

**Comment by vigneshshanmugam** [comment]
> Also, if a user wants to switch the focus back to the original page how does that work? Looking at the code it looks like you'd need to close the popup. Is that right? If so, we should document that.

Not sure if that should be documented, the active window in the step should be screenshoted, otherwise we need to come up with a plan of having multiple screenshots per step if there are more than one active window/pageg.

**Review by andrewvc — APPROVED** [review]
LGTM

**Pull request merged**